### PR TITLE
[client] update documentation with AWS security_group

### PR DESCRIPTION
**Pull request by matthewdeng**
Update Ray Client docs so that it is easier to follow step-by-step how to enable cluster access.

## Why are these changes needed?

The existing Ray Client documentation points to EC2 documentation for enabling inbound access by configuring inbound access rules via security group through the EC2 console. While this does work, for development purposes I found the following two approaches to be easier to follow:
1. Defining Security Group in `cluster.yaml`.
2. SSH Port Forwarding.

## Changelist

1. Add an example to Ray Client doc for configuring `provider.security_group` in `cluster.yaml`  to enable access.
2. Add an `Alternative Approach: Port Forwarding` section.
3. Update cluster config docs to include `provider.security_group` which was previously undocumented.
4. Fix broken Ray Cluster Launcher links.
## Related issue number

n/a

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/config.rst [hunk 1]
```diff
@@ -130,6 +132,20 @@ Provider
             :ref:`project_id <cluster-configuration-project-id>`: str
             :ref:`cache_stopped_nodes <cluster-configuration-cache-stopped-nodes>`: bool
 
+.. _cluster-configuration-security-group-type:
+
+Security Group
+~~~~~~~~~~~~~~
```
Comment (ijrsvt) [inline doc/source/cluster/config.rst:138]: Is this okay as a top-level header?

File: doc/source/cluster/ray-client.rst [hunk 2]
```diff
@@ -99,6 +125,41 @@ Now, connect to the Ray Cluster with the following and then use Ray like you nor
 
    #....
 
+Alternative Approach: Port Forwarding
```
Comment (ijrsvt) [inline doc/source/cluster/ray-client.rst:128]:
```suggestion
Alternative Approach: SSH Port Forwarding
```

**Review by ijrsvt — APPROVED** [review]
Overall LGTM, two small comments.

File: doc/source/cluster/config.rst (hunk 1 above)
Comment (matthewdeng) [reply doc/source/cluster/config.rst:138]: I thought about this too, but saw that the current pattern is to top-level everything in `Custom types`. As an example, (`Node config`, `Node Docker`, `Resources`) are all top-level even though they're only used by `Node Types`. 

Keeping it top-level has the benefit of being able to see "everything available" in the table of contents on the right, but I'm also okay with making it hierarchical if that reads better!

**Review by ckw017 — APPROVED** [review]
lgtm!

File: doc/source/cluster/ray-client.rst [hunk 3]
```diff
@@ -99,6 +125,41 @@ Now, connect to the Ray Cluster with the following and then use Ray like you nor
 
    #....
 
+Alternative Approach: SSH Port Forwarding
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (matthewdeng) [inline doc/source/cluster/ray-client.rst:129]:
```suggestion
Alternative Approach: SSH Port Forwarding
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/source/cluster/config.rst (hunk 1 above)
Comment (ijrsvt) [reply doc/source/cluster/config.rst:138]: LGTM!

**Pull request merged**